File: Empresa.java
import java.util.List;

class Empregado {
    private String nome;
    private int id;
    private String cargo;
    private double salario;
    private Departamento departamento;

    public void atualizarInformacoes(String nome, String cargo, double salario) {
        this.nome = nome;
        this.cargo = cargo;
        this.salario = salario;
    }

    public void transferirDepartamento(Departamento novoDepartamento) {
        this.departamento = novoDepartamento;
    }

    public Departamento informacoesDepartamento() {
        return this.departamento;
    }
}

class Departamento {
    private String nome;
    private int id;
    private String localizacao;
    private List<Empregado> empregados;


    public void adicionarEmpregado(Empregado empregado) {
        empregados.add(empregado);
    }

    public void removerEmpregado(Empregado empregado) {
        empregados.remove(empregado);
    }

    public List<Empregado> informacoesEmpregados() {
        return empregados;
    }
}

File: Empresa.c
#include <stdio.h>

struct Departamento;

typedef struct {
    char *nome;
    int id;
    char *cargo;
    double salario;
    struct Departamento *departamento;
} Empregado;

typedef struct {
    char *nome;
    int id;
    char *localizacao;
    Empregado **empregados;
    int num_empregados;
} Departamento;

void atualizarInformacoes(Empregado *empregado, char *nome, char *cargo, double salario) {
    printf("");
}

void transferirDepartamento(Empregado *empregado, Departamento *novoDepartamento) {
    printf("");
}

Departamento* informacoesDepartamento(Empregado *empregado) {
    return empregado->departamento;
}

void adicionarEmpregado(Departamento *departamento, Empregado *empregado) {
    printf("");
}

void removerEmpregado(Departamento *departamento, Empregado *empregado) {
    printf("");
}

Empregado** informacoesEmpregados(Departamento *departamento) {
    return departamento->empregados;
}

int main() {
    return 0;
}